### PR TITLE
Single Msg YT video support

### LINKED ISSUE #71: Add video support to SM
Uses primary image as holding image with play icon mask. Same behaviour as current component.
Screenshot 2019-08-05 at 08 39 11

- [ ] full screen support

### PR DESCRIPTION
**Pull request by AndyEPhipps**
Type: patch

Fixes #71

### CODE REVIEW

**Comment by LouiaR** [comment]
Not really an issue is it possible to have the video fit his container(mode play) if there is more copy
Screen Shot 2019-09-16 at 09 47 15

File: src/components/Molecules/SingleMessage/SingleMessage.style.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+const PlayButton = styled.button`
+  ${zIndex('high')};
+  cursor: pointer;
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  top: 0;
+  border: 0;
+  background: rgba(0, 0, 0, 0)
+    url(https://www.comicrelief.com/themes/custom/comicrelief/images/copyvideo--play-icon__hover.svg)
```
Comment (gusliedke) [inline src/components/Molecules/SingleMessage/SingleMessage.style.js]: Can we save and use image in `/SingleMessage/assets`?

File: src/components/Molecules/SingleMessage/SingleMessage.style.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    center no-repeat;
+  background-position: center center;
```
Comment (gusliedke) [inline src/components/Molecules/SingleMessage/SingleMessage.style.js]: It's already `center` above, no?!

File: src/components/Molecules/SingleMessage/SingleMessage.style.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  display: ${({ isPlaying }) => (isPlaying ? 'none' : 'block')};
+`;
+
+const Image = styled.div`
+  width: 100%;
+  ${({ doubleImage }) => (doubleImage ? 'height: 100%;' : 'height: 100%;')};
+  ${({ vhFull }) => vhFull && 'height: 100%'};
+
+  z-index: ${({ isPlaying }) => (isPlaying ? '2' : '3')};
```
Comment (gusliedke) [inline src/components/Molecules/SingleMessage/SingleMessage.style.js]: Can we use `${zIndex()};`?

File: src/components/Molecules/SingleMessage/SingleMessage.style.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+`;
+
+const VideoWrapper = styled.div`
+  width: 100%;
+
+  /* Switch z-index layering to place video on top, allow the now-underneath image to maintain height of the wrapper */
+  z-index: ${({ isPlaying }) => (isPlaying ? '3' : '0')};
```
Comment (gusliedke) [inline src/components/Molecules/SingleMessage/SingleMessage.style.js]: same here `${zIndex()};`

File: src/components/Molecules/SingleMessage/SingleMessage.style.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+  // Dynamically loaded via YT API, so can't make it a Styled Component
+  iframe {
+    height: 100%;
+    width: 100%;
+    position: absolute;
+    top: 0;
+    ${({ copyFirst }) =>
+      copyFirst === true ? 'left: auto; right: 0;' : 'left: 0; right: auto;'};
+    z-index: ${({ isPlaying }) => (isPlaying ? '3' : '0')};
```
Comment (gusliedke) [inline src/components/Molecules/SingleMessage/SingleMessage.style.js]: `${zIndex()};`

File: src/components/Molecules/SingleMessage/SingleMessage.style.js (hunk 1 above)
Comment (AndyEPhipps) [reply src/components/Molecules/SingleMessage/SingleMessage.style.js]: Oh yeah, forgot about that

File: src/components/Molecules/SingleMessage/SingleMessage.style.js (hunk 3 above)
Comment (AndyEPhipps) [reply src/components/Molecules/SingleMessage/SingleMessage.style.js]: I tried using in in this context but couldn't seem to get the syntax right; will give it another go

**Comment by AndyEPhipps** [comment]
> Not really an issue is it possible to have the video fit his container(mode play) if there is more copy
> Screen Shot 2019-09-16 at 09 47 15

Nah, YT player that's iframed-in resizes itself automatically, but the 

> Looks nice! Loading gif looks a bit pixelated. Can we get a svg or higher quality version?

yeah for sure, they were more placeholders while I got it all going, but will replace and host both files within the repo 👍

File: src/components/Molecules/SingleMessage/SingleMessage.style.js (hunk 3 above)
Comment (gusliedke) [reply src/components/Molecules/SingleMessage/SingleMessage.style.js]: I think you can use something like ```${({ isPlaying }) => (isPlaying ? zIndex('high') : zIndex('base'))};```

**Comment by AndyEPhipps** [comment]
> Not really an issue is it possible to have the video fit his container(mode play) if there is more copy
> Screen Shot 2019-09-16 at 09 47 15

Nah, YT player that's iframed-in resizes itself automatically via its embedded JS, but the actual video content is always laid out like that. I think the content team have generally been careful with how much copy is.

That said, we actually have 'portrait video' options on CR.com that I overlooked that would suit that kind of layout better, so will do that now; thanks for the reminder!

File: src/components/Molecules/SingleMessage/SingleMessage.style.js (hunk 3 above)
Comment (AndyEPhipps) [reply src/components/Molecules/SingleMessage/SingleMessage.style.js]: Ahh yeah, thats it; thanks!

**Pull request merged**